superdesk/publish/formatters/ninjs_formatter.py

```diff
@@ -426,16 +427,17 @@ def _format_attachments(self, article):
         attachments_service = superdesk.get_resource_service('attachments')
         for attachment_ref in article['attachments']:
             attachment = attachments_service.find_one(req=None, _id=attachment_ref['attachment'])
-            output.append({
-                'id': str(attachment['_id']),
-                'title': attachment['title'],
-                'description': attachment['description'],
-                'filename': attachment['filename'],
-                'mimetype': attachment['mimetype'],
-                'length': attachment.get('length'),
-                'media': str(attachment['media']),
-                'href': '/assets/{}'.format(str(attachment['media'])),
-            })
+            if is_attachment_public(attachment): # don't save internal attachments
+              output.append({
+                  'id': str(attachment['_id']),
+                  'title': attachment['title'],
+                  'description': attachment['description'],
+                  'filename': attachment['filename'],
+                  'mimetype': attachment['mimetype'],
+                  'length': attachment.get('length'),
+                  'media': str(attachment['media']),
+                  'href': '/assets/{}'.format(str(attachment['media'])),
```

Comment:
it's better to use `app.media.url_for_media`

Reply:
you mean instead `href`? cause I didn't change any of this code, I just indented it under the `if`

Reply:
should I change it anyway then?

Reply:
yes pls

Reply:
is this ok?

```
'href': app.media.url_for_media(attachment['media']),
```